bundles/org.eclipse.search/new search/org/eclipse/search2/internal/ui/InternalSearchUI.java

```diff
@@ -228,7 +228,9 @@ private IStatus doRunSearchInForeground(final SearchJobRecord rec, IRunnableCont
 						throw new InvocationTargetException(new CoreException(status));
 					}
 				} catch (OperationCanceledException e) {
-					throw new InterruptedException();
+					InterruptedException interruptedException = new InterruptedException(e.getLocalizedMessage());
+					interruptedException.initCause(e);
+					throw interruptedException;
```

Comment:
In this case, the exception is only misused to transfer an event. This code is inside a try-catch that transforms the exception into a status value, so there is no benefit in adding further information to it:
https://github.com/eclipse-platform/eclipse.platform.ui/blob/6e1920564ae8a68e344f22c3d5c37a2243887e13/bundles/org.eclipse.search/new%20search/org/eclipse/search2/internal/ui/InternalSearchUI.java#L220-L244